docs/design-docs/08-package-variant.md

```diff
@@ -0,0 +1,234 @@
+# Package Variant Controller
+
+* Author(s): @johnbelamaric, @natasha41575
+* Approver: @mortent
+
+## Why
+
+When deploying workloads across large fleets of clusters, it is often necessary
+to modify the workload configuration for a specific cluster. Additionally, those
+workloads may evolve over time with security or other patches that require
+updates. [Configuration as Data](06-config-as-data.md) in general and [Package
+Orchestration](07-package-orchestration.md) in particular can assist in this.
+However, they are still centered around manual, one-by-one hydration and
+configuration of a workload.
+
+This proposal introduces concepts and a set of resources for automating the
+creation and lifecycle management of package variants. These are designed to
+address several different dimensions of scalability:
+- Number of different workloads for a given cluster
+- Number of clusters across which those workloads are deployed
+- Complexity of the organizations deploying those workloads (NOTE: actually this
+  is more the `conditions` stuff, will probably move this out of here and create
+  a separate proposal for that)
+- Changes to those workloads over time
+
+## See Also
```

Comment:
I think this is largely an implementation of the bulk package creation (https://github.com/GoogleContainerTools/kpt/issues/3347) and bulk package updates (https://github.com/GoogleContainerTools/kpt/issues/3243).

Reply:
Updated to reflect this.